src/lib/storage/storage_manager.cpp

```diff
@@ -260,4 +267,316 @@ std::ostream& operator<<(std::ostream& stream, const StorageManager& storage_man
+  chunk_header header;
+  header.row_count = chunk->size();
+  header.segment_offset_ends = segment_offset_ends;
+
+  export_value(header.row_count, file_name);
+
+  for (const auto segment_offset_end : header.segment_offset_ends) {
+    export_value(segment_offset_end, file_name);
+  }
+
+  const auto segment_count = chunk->column_count();
+  for (auto segment_index = size_t{0}; segment_index < segment_count; ++segment_index) {
+    const auto abstract_segment = chunk->get_segment(static_cast<ColumnID>(static_cast<uint16_t>(segment_index)));
+    const auto dict_segment = dynamic_pointer_cast<DictionarySegment<int>>(abstract_segment);
+
+    write_dict_segment_to_disk(dict_segment, file_name);
+  }
+}
+
+void StorageManager::persist_chunks_to_disk(std::vector<std::shared_ptr<Chunk>> chunks, std::string file_name) {
+  // file_lock.acquire();
+
+  auto chunk_segment_offset_ends = std::vector<std::vector<uint32_t>>(StorageManager::CHUNK_COUNT);
+  auto chunk_offset_ends = std::array<uint32_t, StorageManager::CHUNK_COUNT>();
+  auto chunk_ids = std::array<uint32_t, StorageManager::CHUNK_COUNT>();
+
+  auto offset = uint32_t{sizeof(file_header)};
+  for (auto chunk_index = uint32_t{0}; chunk_index < chunks.size(); ++chunk_index) {
+    const auto segment_offset_ends = generate_segment_offset_ends(chunks[chunk_index]);
+    offset += segment_offset_ends.back();
+
+    chunk_segment_offset_ends[chunk_index] = segment_offset_ends;
+    chunk_offset_ends[chunk_index] = offset;
+  }
+  // Fill all offset fields, that are not used with 0s.
+  for (auto rest_chunk_index = chunks.size(); rest_chunk_index < StorageManager::CHUNK_COUNT; ++rest_chunk_index) {
+    chunk_offset_ends[rest_chunk_index] = uint32_t{0};
+  }
+
+  // TODO(everyone): Find, how to get the actual chunk id.
+  for (auto index = uint32_t{0}; index < StorageManager::CHUNK_COUNT; ++index) {
+    chunk_ids[index] = index;
+  }
+
+  file_header fh;
+  fh.storage_format_version_id = STORAGE_FORMAT_VERSION_ID;
+  fh.chunk_count = static_cast<uint32_t>(chunks.size());
+  fh.chunk_ids = chunk_ids;
+  fh.chunk_offset_ends = chunk_offset_ends;
+
+  export_value<file_header>(fh, file_name);
+
+  for (auto chunk_index = uint32_t{0}; chunk_index < chunks.size(); ++chunk_index) {
+    const auto chunk = chunks[chunk_index];
+    write_chunk_to_disk(chunk, chunk_segment_offset_ends[chunk_index], file_name);
+  }
+
+  // file_lock.release();
+}
+
+file_header StorageManager::read_file_header(std::string filename) {
+  file_header file_header;
+  auto fd = int32_t{};
+
+  Assert((fd = open(filename.c_str(), O_RDONLY)) >= 0, "Open error");
+  auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, FILE_HEADER_BYTES, PROT_READ, MAP_PRIVATE, fd, off_t{0}));
+  Assert((map != MAP_FAILED), "Mapping Failed");
+  close(fd);
+
+  file_header.storage_format_version_id = map[0];
+  file_header.chunk_count = map[1];
+
+  const auto fixed_mapped_element_count = 2;
```

Comment:
choose better name, add comment or turn into constant

Reply:
Done.

Reply:
Or remove it.
Not sure what is better.